text/0085-api-keys.md

```diff
@@ -0,0 +1,453 @@
+
+Only the main key allows managing the API keys.
+
+#### Main Key
+
+The main key exists to secure a MeiliSearch instance. As soon as a main key is set via the  `MEILI_MAIN_KEY` environment variable or the `--main-key` CLI option , the endpoint `/keys` is accessible only for the main key holder. It can be seen as a super admin key; It must be shared only with people who have to manage the security of a MeiliSearch instance.
+
+This main key is not an API Key, thus is not stored and fetchable from an API endpoint. It must be seen as a runtime lock that activates the security of MeiliSearch as soon as an instance is launched with it.
+
+At the first launch of MeiliSearch with a main key, MeiliSearch automatically generate two default API keys (See the `GET - /keys` example) to cover many of the most basic needs. It generates a `Default Search API Key` dedicated to the search that can be used on the client-side and a `Default Admin API Key` to manipulate a MeiliSearch instance from a backend side.
+
+If the value of the main key changes, the previously generated `API Keys` do not change or expire. Thus, MeiliSearch does not re-generate the previous generated default API keys.
+
+If the main key is removed at MeiliSearch launch, the previously generated API keys no longer secure the MeiliSearch instance.
+
+If MeiliSearch is launched with the `production` value for the `MEILI_ENV` environement variable or the `--env` CLI option, a main key is mandatory to force the user to secure his instance. If the main key is omitted in that particular case, MeiliSearch launch is aborted and display the `Error: In production mode, the environment variable MEILI_MAIN_KEY is mandatory` error in stdout.
+
+> Note that the main key does not appear on the `/keys` endpoints.
+
+---
+
+#### `API Key` object representation
+
+| field       | type    | description                                      |
+|-------------|---------|--------------------------------------------------|
+| key         | string  | The generated key. **Generated by MeiliSearch**. |
+| actions     | array   | A list of actions permitted for the key. `["*"]` for all actions. See Actions list definition part. |
+| indexes     | array   | A list of indexes permitted for the key. `["*"]` for all indexes. |
+| expiresAt   | string  | Represent the expiration date and time as `ISO-8601` format. `null` equals to no expiration time. |
+| description | string  | A description for the key. `null` if empty. |
+| createdAt   | string  | Represent the date and time as `ISO-8601` format when the API key has been created. **Generated by MeiliSearch** |
+| updatedAt   | string  | Represent the date and time as `ISO-8601` format when the API key has been updated. **Generated by MeiliSearch** |
+
+#### `POST`/`PUT` - `/keys` - API Key object payload definition
```

Comment:
Put or patch then?

Reply:
`patch`, it was a miss on my side 🥲. The route described below uses the verb `patch`.